src/main.rs

```diff
@@ -40,11 +41,18 @@ fn main() -> Result<(), Error> {
 
             println!("{:?}", action);
         }
-
         /*
         TODO: Your implementation of a synthesizer should be here.
         Start with returning an oscillating wave determined by the `time`-variable
         */
+        // added the recommended code and changed it until it stopped giving errors but it is like not exactly working either
+        //by not exactly working i mostly mean bc the audio / ui isnt intergrated yet i cant see/hear it
+        // i aksi dont know the implications of adding a defailt previous phase value but it had to initialize to something in order to compile. also had to define some stuff
+        let mut previous_phase = 1.0;
```

Comment:
yes indeedy there does need to be an initialized previous_phase. How about moving it into a "higher scope" so that the `previous_phase` doesn't get re-created each time this "synth" block loops?